File: lib/clockbar.py
#!/usr/bin/env python3
"""
Short description of this Python module.
Longer description of this module.
This program is free software: you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free Software
Foundation, either version 3 of the License, or (at your option) any later
version.
This program is distributed in the hope that it will be useful, but WITHOUT
ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
You should have received a copy of the GNU General Public License along with
this program. If not, see <http://www.gnu.org/licenses/>.
[AMZYEI]
"""

import gi
import time
import random
from gi.repository import Gtk, GLib

class Clock:
    def __init__(self, window):
        self.window = window
        self.update_time()

    def update_time(self):
        current_time = time.strftime('%H:%M:%S | %d %B %Y ')
        self.window.set_title(f'lazycat - {current_time}')
        self.timeout_id = GLib.timeout_add(1000, self.update_time)

File: include/clockbar.c
/*
 * Clockbar implementation in C using GTK
 * This program updates the window title every second with the current time.
 * 
 * This program is free software: you can redistribute it and/or modify it under
 * the terms of the GNU General Public License as published by the Free Software
 * Foundation, either version 3 of the License, or (at your option) any later version.
 * 
 * This program is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
 * You should have received a copy of the GNU General Public License along with
 * this program. If not, see <http://www.gnu.org/licenses/>.
 * [AMZYEI]
 */

#include <gtk/gtk.h>
#include <time.h>

static gboolean clockbar_update_time(gpointer user_data) {
    GtkWindow *window = GTK_WINDOW(user_data);
    time_t rawtime;
    struct tm *timeinfo;
    char buffer[80];

    time(&rawtime);
    timeinfo = localtime(&rawtime);

    strftime(buffer, sizeof(buffer), "lazycat - %H:%M:%S | %d %B %Y", timeinfo);
    gtk_window_set_title(window, buffer);

    return TRUE; // Continue calling every timeout interval
}

void clockbar_init(GtkWidget *window) {
    // Call clockbar_update_time every 1000 milliseconds (1 second)
    g_timeout_add(1000, clockbar_update_time, window);
}
